source/Halibut/Transport/Protocol/MessageExchangeStream.cs

```diff
@@ -294,9 +294,9 @@ static DataStream FindStreamById(StreamCapture capture, Guid id)
 
         void WriteBsonMessage<T>(T messages)
         {
-            using (var buffer = new BufferedStream(stream, 4096, true))
-            using (var zip = new DeflateStream(buffer, CompressionMode.Compress, true))
-            using (var bson = new BsonDataWriter(zip) { CloseOutput = false })
+            using (var zip = new DeflateStream(stream, CompressionMode.Compress, true))
+            using (var buffer = new BufferedStream(zip, 4096))
```

Comment:
Could you add a comment saying why we need a buffer here, and linking to that blog post please.